src/oatpp/core/async/Processor.cpp

```diff
@@ -189,39 +214,42 @@ void Processor::popTasks() {
 
 void Processor::consumeAllTasks() {
   for(auto& submission : m_taskList) {
-    m_queue.pushBack(submission->createCoroutine(this));
+    m_queue.push_back(submission->createCoroutine(this));
   }
   m_taskList.clear();
 }
 
 void Processor::pushQueues() {
 
-  oatpp::collection::FastQueue<CoroutineHandle> tmpList;
+  std::vector<CoroutineHandle*> tmpList;
 
   {
     std::lock_guard<oatpp::concurrency::SpinLock> lock(m_taskLock);
     consumeAllTasks();
-    oatpp::collection::FastQueue<CoroutineHandle>::moveAll(m_pushList, tmpList);
+    tmpList = std::move(m_pushList);
   }
 
-  while(tmpList.first != nullptr) {
-    addCoroutine(tmpList.popFront());
+  for (CoroutineHandle* coroutine : tmpList) {
+    addCoroutine(coroutine);
   }
 
 }
 
 bool Processor::iterate(v_int32 numIterations) {
-
+  std::lock_guard<oatpp::concurrency::SpinLock> lg(m_threadLock);
   pushQueues();
 
-  for(v_int32 i = 0; i < numIterations; i++) {
+  for(v_int32 i = 0; i < numIterations && m_running; i++) {
 
-    auto CP = m_queue.first;
-    if (CP == nullptr) {
+    if (m_queue.empty()) {
       break;
     }
+
+    std::swap(m_queue.front(), m_queue.back());
```

Comment:
This was changed, because we previously had a list where removing elements from the front would not copy/move all other elements. Since we now have a vector we simply swap the first element with the last and remove it from the back of the vector without touching the other elements.